### ISSUE TITLE
[C#] [Windows] Project Platform Target only works on x86 ?

### ISSUE THREAD

**Issue by IgorMundstein** [mz-automation/libiec61850#26]
Hi,

I realized that if the project platform target is not x86.

![image](https://user-images.githubusercontent.com/3512138/32302671-06194e12-bf4b-11e7-8096-f7a54f30ac7f.png)

I get the exception:

![image](https://user-images.githubusercontent.com/3512138/32302657-f9865d52-bf4a-11e7-8ac1-0a66293d1bab.png)

`System.BadImageFormatException was unhandled
  HResult=-2147024885
  Message=An attempt was made to load a program with an incorrect format. (Exception from HRESULT: 0x8007000B)
  Source=iec61850dotnet
  StackTrace:
       at IEC61850.Client.IedConnection.IedConnection_create()
       at IEC61850.Client.IedConnection..ctor() in C:\libiec61850-master\dotnet\IEC61850forCSharp\IEC61850ClientAPI.cs:line 410
       at example1.MainClass.Main(String[] args) in C:\libiec61850-master\dotnet\example1\Main.cs:line 13
       at System.AppDomain._nExecuteAssembly(RuntimeAssembly assembly, String[] args)
       at System.AppDomain.ExecuteAssembly(String assemblyFile, Evidence assemblySecurity, String[] args)
       at Microsoft.VisualStudio.HostingProcess.HostProc.RunUsersAssembly()
       at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
       at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
       at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
       at System.Threading.ThreadHelper.ThreadStart()
  InnerException: 
`

**I created C++ library by command: cmake -G "Visual Studio 14" C:\libiec61850-master**

**Am I doing something wrong? Do I have to configure some platform variable?**

**Comment by IgorMundstein** [comment]
Hi,

![image](https://user-images.githubusercontent.com/3512138/32444449-dbd24238-c2e9-11e7-8bbe-b7070b1cf76e.png)

I removed the parameter "/machine:X86" and I was able to compile the DLL in 64 bit. The problem has apparently been solved.

This can be done? Will it generate any future problems?

**Comment by mzillgith** [comment]
Hello, I think the only problem would be that the DLL can only be used on a 64 bit system. But despite of that I think it is ok.

**Comment by IgorMundstein** [comment]
@mzillgith 

OK. I will test and keep you informed.

Suggestion: It would be interesting to create a new 64-bit configuration for CMAKE. Something like that: Debug;Release;MinSizeRel;RelWithDebInfo;**Release64**)

**Comment by mzillgith** [comment]
@IgorMundstein 

After some research I am not sure if it is easily possible to add a separate 64 bit target in a platform independent way. I think the easiest way to build for 64 bit is to select the generator with the "Win64" option. E.g.:
cmake -G "Visual Studio 14 2015 Win64"

**Comment by IgorMundstein** [comment]
@mzillgith 

Ok. Thx!

**Comment by azlanms** [comment]
I had similar issues of getting System.DllNotFoundException and System.BadImageExceptionFormat errors while deploying my application on another PC.

For the former case, I followed the recommendation given in https://libiec61850.com/libiec61850/documentation/iec-61850-client-tutorial/#comment-83779. Then I check the iec61850.dll dependency and discovered that it requires vcruntim140d.dll and ucrtbased.dll since I was developing under debug mode. I copied these two files into the PC but it lead to the latter error message.

Later I discovered that the vcruntim140d.dll and ucrtbased.dll are from older Visual Studio. Then I copied the correct versions to the PC and the problems were resolved.
